### ISSUE TITLE
var patterns should be assigned in multi-label switch sections

### ISSUE THREAD

**Issue by alrz** [dotnet/roslyn#23140]
**Version Used**: 7.1

**Steps to Reproduce**:
```cs
void M(int i) {
  switch (i) {
    case 1:
    case 2:
    case var x:
      var y = x;
      break;
   }
}
```
**Expected Behavior**: compile with no error

**Actual Behavior**: error: use of unassigned variable `x`

**Comment by jcouv** [comment]
Tagging @gafter 
Personally, I don't see the value. Just remove `case 1:` and `case 2:`.

**Comment by alrz** [comment]
Imagine there are other sections as well, and you're switching on a function. the only workaround is to use a local here, but logically I'd expect this to work because var pattern is not meant to fail in any ways.

**Comment by gafter** [comment]
@alrz According to the specification, the lexically first matching case pattern is the one that determines the code to be executed. No subsequent patterns are considered in determining the matching switch section. Since `case 1` causes `case var x` to be bypassed, the variable `x` is not definitely assigned.

This is by design.
